modules/geo-layers/src/tile-layer/utils.js

```diff
@@ -141,9 +186,18 @@ export function getTileIndices({viewport, maxZoom, minZoom, zRange, extent, tile
   if (Number.isFinite(maxZoom) && z > maxZoom) {
     z = maxZoom;
   }
+  let transformedExtent = extent;
+  if (modelMatrix && extent && !viewport.isGeospatial) {
+    transformedExtent = transformBox(extent, modelMatrix);
+  }
   return viewport.isGeospatial
     ? getOSMTileIndices(viewport, z, zRange, extent || DEFAULT_EXTENT)
-    : getIdentityTileIndices(viewport, z, extent || DEFAULT_EXTENT);
+    : getIdentityTileIndices(
+        viewport,
+        z,
+        transformedExtent || DEFAULT_EXTENT,
+        modelMatrix || new Matrix4()
```

Comment:
I suggest you avoid assigning a default value here and check for null everywhere before you use the matrix. That way we can be confident that for existing use cases there will be no behavioral change/perf hit.